python/pyspark/testing/utils.py

```diff
@@ -221,7 +221,13 @@ def check_error(
         )
 
 
-def assertDataFrameEqual(df: DataFrame, expected: DataFrame, check_row_order: bool = False):
+def assertDataFrameEqual(
+    df: DataFrame,
+    expected: DataFrame,
+    check_row_order: bool = False,
```

Comment:
While we're here, I think we should remove `check_row_order`. We don't really guarantee the order in Spark Dataframe.

Reply:
we decided to keep (renamed to `checkRowOrder`), with added documentation clarifying that ordering isn't guaranteed unless df is explicitly sorted